### PR TITLE
add an automated CI workflow to sync ODH-notebooks with latest codefl…

### PR DESCRIPTION
**Pull request by abhijeet-dhumal**
# Issue link

[RHOAIENG-3756](https://issues.redhat.com/browse/RHOAIENG-3756)

# What changes have been made

Add step to sync ODH notebooks with Codeflare-SDK release by creating automated PR for Project-Codeflare-notebooks repository with latest Cddeflare-SDK release version

Related PR : https://github.com/project-codeflare/codeflare-operator/pull/482

# Verification steps

A test run in fork results in the following automated PR changes :
[Successful test build ](https://github.com/abhijeet-dhumal/codeflare-sdk/actions/runs/8065068305/job/22030161646#logs)
[Generated automated test PR](https://github.com/sutaakar/notebooks/pull/10)
[Incase Codeflare-SDK version doesn't exist](https://github.com/abhijeet-dhumal/codeflare-sdk/actions/runs/8063089657/job/22024208957#step:6:82)

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: .github/workflows/odh-notebooks-sync.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  REPO_NAME: notebooks
+  GITHUB_TOKEN: ${{ secrets.CODEFLARE_MACHINE_ACCOUNT_TOKEN }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ env.BRANCH_NAME }}
```
Comment (sutaakar) [inline .github/workflows/odh-notebooks-sync.yml:35]: Makes me think, what is SDK repository used for in this workflow?

Comment (abhijeet-dhumal) [reply .github/workflows/odh-notebooks-sync.yml:35]: Yes right, I think this step is not required ~

As we are clonning project-codeflare's/notebooks branch freshly after checkout step and again creating new branch using checkout to make changes? Yeah it doesn't makes sense
For example : 
Here by default I'm using `project-codeflare/notebooks` as a base to make changes and create PR against `ODH/notebooks` in this workflow steps

Thank you @sutaakar for pointing this out!
Let me test it again by removing this checkout step!

**Review by Srihari1192 — APPROVED** [review]
lgtm

**Review by sutaakar — APPROVED** [review]
/lgtm

**Comment by Fiona-Waters** [comment]
Great job! @abhijeet-dhumal 
/approve

**Pull request merged**